### ISSUE TITLE
Error with normalization

### ISSUE THREAD

**Issue by nikolayaivanov** [hansenlab/minfi#200]
Hi,
I'm encountering an error with normalization:
```
GRset.quantile=preprocessQuantile(RGset)
[preprocessQuantile] Mapping to genome.
Loading required package: IlluminaHumanMethylation450kanno.ilmn12.hg19
[preprocessQuantile] Fixing outliers.
[preprocessQuantile] Quantile normalizing.
Error in normalize.quantiles(mat[Index2, ]) :
  ERROR; return code from pthread_create() is 22
```
Here's the sessionInfo. Of note, the BLAS that I was using was `openblas-0.3.7-h6e990d7_3`
```
> sessionInfo()
R version 3.6.1 (2019-07-05)
Platform: x86_64-conda_cos6-linux-gnu (64-bit)
Running under: Red Hat Enterprise Linux Server release 6.3 (Santiago)

Matrix products: default
BLAS/LAPACK: /pbtech_mounts/homes027/nai2008/anaconda3/lib/R/lib/libRblas.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C
 [9] LC_ADDRESS=C               LC_TELEPHONE=C
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets
[8] methods   base

other attached packages:
 [1] IlluminaHumanMethylation450kanno.ilmn12.hg19_0.6.0
 [2] calibrate_1.7.5
 [3] MASS_7.3-51.5
 [4] IlluminaHumanMethylation450kmanifest_0.4.0
 [5] minfi_1.32.0
 [6] bumphunter_1.28.0
 [7] locfit_1.5-9.1
 [8] iterators_1.0.12
 [9] foreach_1.4.7
[10] Biostrings_2.54.0
[11] XVector_0.26.0
[12] SummarizedExperiment_1.16.1
[13] DelayedArray_0.12.2
[14] BiocParallel_1.20.1
[15] matrixStats_0.55.0
[16] Biobase_2.46.0
[17] GenomicRanges_1.38.0
[18] GenomeInfoDb_1.22.0
[19] IRanges_2.20.1
[20] S4Vectors_0.24.1
[21] BiocGenerics_0.32.0
```
This thread https://support.bioconductor.org/p/122925/ indicates that this issue appears to be occurring with openblas version  >= 0.3.4. This issue was resolved when I downgraded from `openblas-0.3.7-h6e990d7_3` to `openblas-0.3.3-h9ac9557_1001`.

The same error occurred with `preprocessFunnorm`.

Any advice?
Thank you!

Best,
Nikolay

**Comment by kasperdanielhansen** [comment]
Skimming this, it sounds like you have been using a BLAS version which doesn't work. I don't think this is a mini problem.

**Comment by nikolayaivanov** [comment]
Hi Dr. Hansen, 
Sorry I meant to write that this issue was **not** resolved when I downgraded from openblas-0.3.7-h6e990d7_3 to openblas-0.3.3-h9ac9557_1001.

**Comment by nikolayaivanov** [comment]
Hi Dr. Hansen,
Sorry I meant to write that this issue was not resolved when I downgraded from openblas-0.3.7-h6e990d7_3 to openblas-0.3.3-h9ac9557_1001.

--
Nikolay A. Ivanov
Weill Cornell Medicine
MD Candidate | Class of 2021
T 240.855.4027 | nai2008@med.cornell.edu<mailto:jhg2004@med.cornell.edu>







On Feb 18, 2020, at 1:21 PM, Kasper Daniel Hansen <notifications@github.com<mailto:notifications@github.com>> wrote:


Closed #200<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_hansenlab_minfi_issues_200&d=DwMCaQ&c=lb62iw4YL4RFalcE2hQUQealT9-RXrryqt9KZX2qu2s&r=fwsTWRy9v0yJvx4GNoot-tQwYdIn-e1NnSFAvQOEnNE&m=mQkzJGw5HM2BMDbwy2v7pC-PwcvZQL4EG_4MIHJdjCY&s=yxg3kvZl8awE2zxt4CS4Knvo6m0o6olcCpL_WmRUJAg&e=>.

**Comment by nikolayaivanov** [comment]
This fixed the issue:
https://support.bioconductor.org/p/122925/

I downloaded the Source Package of `preprocessCore` from https://www.bioconductor.org/packages/release/bioc/html/preprocessCore.html,
and then manually installed it, disabling threading:

`R CMD INSTALL --configure-args="--disable-threading" preprocessCore_1.48.0.tar.gz`

Now, `preprocessQuantile()` and `preprocessFunnorm()` run without throwing an error.
